python/ray/actor.py

```diff
@@ -129,34 +127,112 @@ def annotate_method(method):
     return annotate_method
 
 
+class ActorMethodShell:
```

Comment:
```suggestion
class ActorMethodMetadata:
```

better name IMO (after reading your docstring)